splitting_data_by_type_year.py

```diff
@@ -0,0 +1,113 @@
+import os
+import pandas as pd
+import logging
+from typing import List, Dict
+import re
+from datetime import datetime
+
+log_directory: str = r'./logs'
+os.makedirs(log_directory, exist_ok=True)
+logging.basicConfig(
+    filename=os.path.join(log_directory, 'data_split.log'),
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s',
+    encoding='utf-8'
+)
+ 
+def check_folder_exists(folder_path: str) -> None:
+    '''Check if folder exists'''
+    if not os.path.exists(folder_path):
+        logging.error(f'Folder {folder_path} does not exist')
+        raise FileNotFoundError(f'Folder {folder_path} does not exist')
```

Comment:
FYI: https://www.programiz.com/python-programming/user-defined-exception

Reply:
Sure, but I do not plan to use OOP in this project.